onyo/commands/tree.py

```diff
@@ -18,18 +18,26 @@ def sanitize_directories(directories: list, opdir: str) -> list[str]:
     """
     dirs = []
     error_path = []
+    no_path = []
 
     for d in directories:
-        full_path = Path(opdir, d)
+        full_path = Path(repo.opdir, d).resolve()
+        if not full_path.is_relative_to(repo.root):
+            error_path.append(d)
+            continue
 
         if full_path.is_dir():
-            dirs.append(full_path.relative_to(opdir))
+            dirs.append(full_path.relative_to(repo.opdir))
         else:
-            error_path.append(d)
+            no_path.append(d)
 
-    if error_path:
-        print('The following paths are not directories:\n' + '\n'.join(error_path),
-              file=sys.stderr)
+    if error_path or no_path:
+        if error_path:
+            print('The following paths are not in the Repository:\n' +
```

Comment:
Nit: `Repository` should not be capitalized.

In general, I prefer when errors are explicit about what is wrong.

The issue is not "that the path is not in the repo". It's that all paths must be in the repo.

Reply:
Thanks, I thought about a few ways of stating this more explicitly, please let me know what you think :)